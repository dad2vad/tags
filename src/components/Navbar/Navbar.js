import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import Icon from '../Icon';
import Link from '../Link';
import config from '../../config.json';
import logo from '../../assets/logo.svg';

const Navbar = ({ onNavToggle, isMenuOpened }) => {
  const toggleClass = classNames('nav-toggle', {
    'is-active': isMenuOpened,
  });
  const navClass = classNames('nav-right', 'nav-menu', {
    'is-active': isMenuOpened,
  });
  return (
    <nav className="nav">
      <div className="container is-fullhd">
        <div className="nav-left">
          <div className="nav-item">
            <img src={logo} alt={config.title} />
            {config.title}
          </div>
        </div>
        <span onClick={onNavToggle} className={toggleClass}>
          <span />
          <span />
          <span />
        </span>
        <div className={navClass}>
          <Link url="https://github.com/Landish/frameworks.now.sh" className="nav-item">
            <Icon name="github" />
            Source Code
          </Link>
          <Link
            url="https://github.com/Landish/frameworks.now.sh/blob/master/CHANGELOG.md"
            className="nav-item"
          >
            <span className="tag is-white is-small">v{config.version}</span>
          </Link>
        </div>
      </div>
    </nav>
  );
};

Navbar.propTypes = {
  onNavToggle: PropTypes.func.isRequired,
  isMenuOpened: PropTypes.bool,
};

Navbar.defaultProps = {
  isMenuOpened: false,
};

export default Navbar;
