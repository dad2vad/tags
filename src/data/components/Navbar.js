export default {
  name: 'Navbar',
  supports: {
    amaze: 'http://amazeui.github.io/docs/en/css/topbar/',
    blaze: 'http://blazecss.com/components/navs/',
    bootstrap3: 'https://getbootstrap.com/docs/3.3/components/#navbar',
    bootstrap4: 'https://getbootstrap.com/docs/4.0/components/navbar/',
    bulma: 'http://bulma.io/documentation/components/navbar/',
    foundation5: 'http://foundation.zurb.com/sites/docs/v/5.5.3/components/topbar.html',
    foundation6: 'http://foundation.zurb.com/sites/docs/top-bar.html',
    gumby: 'https://gumbyframework.com/docs/ui-kit/#!/navigation',
    ink: 'http://ink.sapo.pt/ui-elements/navigation/',
    materialize: 'http://materializecss.com/navbar.html',
    metro: 'https://metroui.org.ua/appbar.html',
    mini: 'http://minicss.org/navigation#header',
    mui: 'https://www.muicss.com/docs/v1/css-js/appbar',
    office: 'http://dev.office.com/fabric#/components/commandbar',
    uikit2: 'https://getuikit.com/v2/docs/navbar.html',
    uikit3: 'https://getuikit.com/docs/navbar',
    semantic: 'http://semantic-ui.com/collections/menu.html',
    spectre: 'https://picturepan2.github.io/spectre/layout.html#navbar',
  },
};
