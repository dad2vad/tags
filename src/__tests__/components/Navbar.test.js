import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../../components/Navbar';

describe('Navbar', () => {
  it('renders correctly', () => {
    const component = shallow(<Navbar isMenuOpened={false} onNavToggle={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});
