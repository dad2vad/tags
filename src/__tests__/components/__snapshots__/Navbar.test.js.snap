// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<nav
  className="nav"
>
  <div
    className="container is-fullhd"
  >
    <div
      className="nav-left"
    >
      <div
        className="nav-item"
      >
        <img
          alt="Comparison of Front-End Frameworks"
          src="logo.svg"
        />
        Comparison of Front-End Frameworks
      </div>
    </div>
    <span
      className="nav-toggle"
      onClick={[Function]}
    >
      <span />
      <span />
      <span />
    </span>
    <div
      className="nav-right nav-menu"
    >
      <Link
        className="nav-item"
        url="https://github.com/Landish/frameworks.now.sh"
      >
        <Icon
          name="github"
        />
        Source Code
      </Link>
      <Link
        className="nav-item"
        url="https://github.com/Landish/frameworks.now.sh/blob/master/CHANGELOG.md"
      >
        <span
          className="tag is-white is-small"
        >
          v
          1.3.0
        </span>
      </Link>
    </div>
  </div>
</nav>
`;
